pkg/platform/runtime/setup/setup.go

```diff
@@ -289,7 +324,8 @@ func (s *Setup) installFromBuildLog(buildResult *model.BuildResult, artifacts ma
 		return err
 	}
 
-	wp.StopWait()
+	// wait for the build artifacts to be processed
+	wg.Wait()
```

Comment:
Why was this necessary?

Reply:
It is an unlikely situation, but likely enough such that I ran into it during testing:

Here is how it was.


```
Thread A                        Thread B
receive last artifact download (line 278 / 307)
                                       buildLog.Wait() returns (line 288/323)
                                       wp.StopWait() (line 292) closes the workerpool channel
wp.Submit() (line 279/310) causes a "closed channel"-panic.
```

The new version ensures that `wp.StopWait()` is called after all `wp.Submit()` calls are issued.

Reply:
Maybe add some comments explaining this so someone doesn't accidentally "clean this up" (ie. me.. :p)